.github/workflows/ci.yml

```diff
@@ -0,0 +1,101 @@
+name: CI
+
+on:
+  pull_request_target:
+    types: [opened, synchronize, reopened]
```

Comment:
Have we considered having jobs in individual workflow files? I would love that for code readability. Obviously, when actions depend on each other it will become more difficult, but here these are mostly separate/self-contained.

Reply:
I think eventually we can go to that, but I want to play with it to understand how to best do this.  I have some ideas with `workflow_calls` that could be a nice option.